src/ImageSearchModal.jsx

```diff
@@ -70,28 +70,28 @@ export class ImageSearchModal extends React.Component {
 
         this.setState({ searchInProgress: true });
 
-        varlink.connect(utils.getAddress(this.state.isSystem), this.state.isSystem)
-                .then(connection => {
-                    this.activeConnection = connection;
-
-                    connection.call("io.podman.SearchImages", { query: this.state.imageIdentifier })
-                            .then(reply => {
-                                if (this._isMounted)
-                                    this.setState({ imageList: reply.results || [], searchInProgress: false, searchFinished: true });
-                            })
-                            .catch(ex => {
-                                // We expect new searches to close the connection for ongoing searches
-                                if (ex.error == 'ConnectionClosed')
-                                    return;
-
-                                if (this._isMounted) {
-                                    this.setState({
-                                        searchInProgress: false,
-                                        dialogError: _("Failed to search for new images"),
-                                        dialogErrorDetail: cockpit.format("$0: $1", ex.error, ex.parameters && ex.parameters.reason)
-                                    });
-                                }
-                            });
+        this.activeConnection = rest.connect(client.getAddress(this.state.isSystem), this.state.isSystem);
```

Comment:
This opens a new connection for every search, but still remembers it in `this`. Is that something that should be persistent (and is expensive enough to re-use)? Then it should check if it's already initialized. Or is it cheap enough to really re-do every time and should not be remembered outside of that search trigger callback?

Reply:
You mean that we would do every `this.activeConnection.call(...)` on the same `activeConnection`?
I think there might be a problem, if one `call()` does not yet reply, but you want to do another `call()`. In this case we close previous active connection if it did not yet reply, and create a new one.
It is the same approach that we had before. I haven't really study if it is cheap or expensive to do, but I haven't noticed any significant delay.

Reply:
So doing a new `rest.connect()` for each call sounds safer indeed. But then it shouldn't be kept in an object member variable, but just be a local one.

Reply:
But we need to `.close()` the previous connection when a new one is triggered, thus it needs to be saved on some higher level. As member it seems cleaner to me, than having top level variable.

Reply:
OK, I see it now, the close() isn't part of this diff. That's certainly rather unusual (closing the connection would make more sense right after getting the respose), but ok for now. Thanks!